### PR TITLE
Bluetooth: controller: Fix encryption issues regarding rejection and termination

### PR DESCRIPTION
**Pull request by thoh-ot**
EBQ test run passes (https://github.com/zephyrproject-rtos/zephyr/issues/43435):
- LL/SEC/PER/BI-04-C
- LL/SEC/PER/BI-03-C
- LL/SEC/PER/BI-01-C
- LL/SEC/CEN/BV-12-C
- LL/SEC/CEN/BV-13-C
- LL/SEC/CEN/BV-03-C
- LL/SEC/CEN/BV-14-C
- LL/SEC/PER/BI-05-C
- LL/SEC/PER/BV-02-C
- LL/SEC/PER/BV-01-C

Builds on top of the fixes in
https://github.com/zephyrproject-rtos/zephyr/pull/44275
https://github.com/zephyrproject-rtos/zephyr/pull/44014
https://github.com/zephyrproject-rtos/zephyr/pull/44003

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_llcp_enc.c [hunk 1]
```diff
@@ -406,6 +406,14 @@ static void lp_enc_st_wait_rx_enc_rsp(struct ll_conn *conn, struct proc_ctx *ctx
 		/* Pause Rx data */
 		ull_conn_pause_rx_data(conn);
 		lp_enc_store_s(conn, ctx, pdu);
+
+		/* After the Central receives the LL_ENC_RSP PDU in response,
+		 * only PDUs required by this procedure are expected;
+		 * to achieve that even "unwanted" PDU's end up in this FSM,
+		 * enable the greedy RX behaviour.
+		 */
```
Comment (erbr-ot) [inline subsys/bluetooth/controller/ll_sw/ull_llcp_enc.c:414]: Suggestion for reformulation:
	/* After the Central has received the LL_ENC_RSP PDU,
		 * only PDUs related to this procedure are valid, and invalids should 
		 * result in disconnect.
		 * to achieve this enable the greedy RX behaviour, such that
		 * all PDU's end up in this FSM. 
		 */

**Review by erbr-ot — DISMISSED** [review]
Looks real solid to me. Only a minor comment re. clarification in a comment

File: subsys/bluetooth/controller/ll_sw/ull_llcp_enc.c (hunk 1 above)
Comment (thoh-ot) [reply subsys/bluetooth/controller/ll_sw/ull_llcp_enc.c:415]: ACK

**Comment by thoh-ot** [comment]
Will wait with the rebase until https://github.com/zephyrproject-rtos/zephyr/pull/44275 is merged

**Comment by thoh-ot** [comment]
Rebased

**Review by kruithofa — APPROVED** [review]

**Review by erbr-ot — APPROVED** [review]
LGTM

**Review by cvinayak — APPROVED** [review]

**Pull request merged**